### PR TITLE
Molecule fixes [idrac_bios | idrac_export_scp | idrac_attributes]

### PR DESCRIPTION
**Pull request by singh-lovepreet1**
# Description
Molecule Fixes for the below roles:
 1. idrac_bios
 2. idrac_attributes
 3. idrac_export_server_config_profile

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- Test Pull Request

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

### CODE REVIEW

**Comment by ansible-collections-svc** [comment]
Can one of the admins verify this patch?

**Review by rajshekarp87 — DISMISSED** [review]

File: roles/idrac_bios/molecule/clear_pending_attributes/prepare.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    - name: Fetch Bios Jobs Data
+      when: idrac_bios_uri_data.json.Members | length > 0
+      ansible.builtin.set_fact:
+        idrac_bios_jobs_items: "{{ idrac_bios_uri_data.json.Members | json_query(query) }}"
+      vars:
+        query: "[?JobType=='BIOSConfiguration' && JobState=='Scheduled' || JobState=='Scheduling' ]"
+      no_log: true
+
+    - name: Block for creating a bios job as a pre-requisite
```
Comment (ABHISHEK-SINHA10) [inline roles/idrac_bios/molecule/clear_pending_attributes/prepare.yml:26]: Good to see 👍  using prepare.yml for pre-requisite.

File: roles/idrac_bios/molecule/clear_pending_attributes/prepare.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+---
+- name: Converge file to update attributes with apply time as Immediate
```
Comment (ABHISHEK-SINHA10) [inline roles/idrac_bios/molecule/clear_pending_attributes/prepare.yml:2]: Update the name, it's prepare.yml not converge.yml, check other places too

File: roles/idrac_bios/molecule/update_attributes_immediate/cleanup.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+---
+- name: Converge file to update attributes with apply time as Immediate
+  hosts: all
+  gather_facts: false
+  vars:
+    idrac_ip: "{{ lookup('env', 'IDRAC_IP') }}"
```
Comment (ABHISHEK-SINHA10) [inline roles/idrac_bios/molecule/update_attributes_immediate/cleanup.yml]: No use of vars mentioned in this file, if not needed remove it. Check name too, need update.

File: roles/idrac_bios/molecule/update_attributes_immediate/prepare.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
+---
+- name: Converge file to update attributes with apply time as Immediate
```
Comment (ABHISHEK-SINHA10) [inline roles/idrac_bios/molecule/update_attributes_immediate/prepare.yml]: same here, check name

File: roles/idrac_bios/molecule/update_attributes_immediate/prepare.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+      vars:
+        url: https://{{ idrac_ip }}:{{ idrac_port }}/redfish/v1/Systems/System.Embedded.1/Bios
+
+    - name: Fetch the existing boot mode
+      ansible.builtin.set_fact:
+        current_boot_mode: "{{ idrac_bios_uri_data.json.Attributes.BootMode }}"
+
+    - name: Set the boot mode
+      ansible.builtin.set_fact:
+        set_boot_mode: "{{ 'Bios' if (current_boot_mode == 'Uefi') else 'Uefi' }}"
```
Comment (ABHISHEK-SINHA10) [inline roles/idrac_bios/molecule/update_attributes_immediate/prepare.yml]: Good 👍 for using if else statement inside Jinja template

File: roles/idrac_bios/molecule/update_attributes_immediate_with_jobwait/cleanup.yml [hunk 6]
```diff
@@ -0,0 +1,20 @@
+---
+- name: Converge file to update attributes with apply time as Immediate
+  hosts: all
+  gather_facts: false
+  vars:
+    idrac_ip: "{{ lookup('env', 'IDRAC_IP') }}"
```
Comment (ABHISHEK-SINHA10) [inline roles/idrac_bios/molecule/update_attributes_immediate_with_jobwait/cleanup.yml]: check if vars needed in file

File: roles/idrac_bios/molecule/update_attributes_immediate_with_jobwait/prepare.yml [hunk 7]
```diff
@@ -0,0 +1,29 @@
+---
+- name: Converge file to update attributes with apply time as Immediate
```
Comment (ABHISHEK-SINHA10) [inline roles/idrac_bios/molecule/update_attributes_immediate_with_jobwait/prepare.yml]: Update proper name

File: roles/idrac_bios/molecule/update_attributes_immediate_with_jobwait/prepare.yml [hunk 8]
```diff
@@ -0,0 +1,29 @@
+---
```
Comment (ABHISHEK-SINHA10) [inline roles/idrac_bios/molecule/update_attributes_immediate_with_jobwait/prepare.yml]: if prepare.yml is common for all the scenario, have a look to molecule documentation, we can have common file throughout the scenario.

https://ansible.readthedocs.io/projects/molecule/configuration/#molecule.provisioner.ansible.Ansible

image

**Review by ShivamSh3 — DISMISSED** [review]

File: roles/idrac_bios/molecule/clear_pending_attributes/prepare.yml (hunk 2 above)
Comment (singh-lovepreet1) [reply roles/idrac_bios/molecule/clear_pending_attributes/prepare.yml:2]: Updated.

File: roles/idrac_bios/molecule/update_attributes_immediate/cleanup.yml (hunk 3 above)
Comment (singh-lovepreet1) [reply roles/idrac_bios/molecule/update_attributes_immediate/cleanup.yml]: Removed and updated the file

File: roles/idrac_bios/molecule/update_attributes_immediate/prepare.yml (hunk 4 above)
Comment (singh-lovepreet1) [reply roles/idrac_bios/molecule/update_attributes_immediate/prepare.yml]: Reviewed and updated accordingly

File: roles/idrac_bios/molecule/update_attributes_immediate_with_jobwait/cleanup.yml (hunk 6 above)
Comment (singh-lovepreet1) [reply roles/idrac_bios/molecule/update_attributes_immediate_with_jobwait/cleanup.yml]: Reviewed and removed unnecessary variables

File: roles/idrac_bios/molecule/update_attributes_immediate_with_jobwait/prepare.yml (hunk 7 above)
Comment (singh-lovepreet1) [reply roles/idrac_bios/molecule/update_attributes_immediate_with_jobwait/prepare.yml]: Updated

File: roles/idrac_bios/molecule/update_attributes_immediate_with_jobwait/prepare.yml (hunk 8 above)
Comment (singh-lovepreet1) [reply roles/idrac_bios/molecule/update_attributes_immediate_with_jobwait/prepare.yml]: Updated. Now, a single file for prepare and cleanup is there.

**Review by ABHISHEK-SINHA10 — DISMISSED** [review]

**Review by ShivamSh3 — DISMISSED** [review]

File: roles/idrac_bios/molecule/clear_pending_attributes/prepare.yml [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+          vars:
+            url: https://{{ idrac_ip }}:{{ idrac_port }}/redfish/v1/Systems/System.Embedded.1/Bios
+
+        - name: Fetch the existing boot mode
+          ansible.builtin.set_fact:
+            current_boot_mode: "{{ idrac_bios_uri_data.json.Attributes.BootMode }}"
+
+        - name: Set the boot mode
+          ansible.builtin.set_fact:
+            set_boot_mode: "{{ 'Bios' if (current_boot_mode == 'Uefi') else 'Uefi' }}"
```
Comment (Jennifer-John) [inline roles/idrac_bios/molecule/clear_pending_attributes/prepare.yml]: can these vars not be defined as idrac_bios_set_boot_mode, idrac_bios_current_boot_mode.. similar to the others?

Comment (singh-lovepreet1) [reply roles/idrac_bios/molecule/clear_pending_attributes/prepare.yml]: Not possible as that is giving sonar issues if we rename it.

**Review by jagadeeshnv — DISMISSED** [review]

File: roles/idrac_bios/molecule/default/converge.yml [hunk 10]
```diff
@@ -3,22 +3,24 @@
...
-        name: idrac_bios
-      vars:
-        hostname: "randomHostname"
-        username: "{{ lookup('env', 'IDRAC_USER') }}"
-        password: "{{ lookup('env', 'IDRAC_PASSWORD') }}"
-        validate_certs: false
-        reset_bios: true
-      ignore_unreachable: true
+    ########## Below snippet is commented because of traceback error  ########
+    # - name: Perform reset bios with invalid hostname
```
Comment (ABHISHEK-SINHA10) [inline roles/idrac_bios/molecule/default/converge.yml]: If you have raised JIT, please mention in comment if it's raising traceback error.

**Review by Kritika-Bhateja-03 — DISMISSED** [review]

File: roles/idrac_bios/molecule/default/converge.yml (hunk 10 above)
Comment (singh-lovepreet1) [reply roles/idrac_bios/molecule/default/converge.yml]: Updated!

**Review by ABHISHEK-SINHA10 — APPROVED** [review]

**Review by sachin-apa — APPROVED** [review]

**Review by jagadeeshnv — APPROVED** [review]

**Review by Kritika-Bhateja-03 — APPROVED** [review]

**Review by Jennifer-John — APPROVED** [review]

**Pull request merged**